### ISSUE TITLE
Geocode 1989 data

### ISSUE THREAD

**Issue by kbuzard** [kbuzard/labs#55]
In July 2022, @Kirs10-Riley started trying to apply Antonio's workflow to the 1989 version of the Cattell directory, building off the work Kelly did during Summer 2021.

**Comment by kbuzard** [comment]
I'm copying @Kirs10-Riley's reflection from her daily log here to keep everything about this task in one place:

### Wednesday, July 27

Hours worked on this task today: 3
Total hours worked on this task
Reflection: I created a folder in the Python Script folder called 1989 where all of the copied scripts are, but I am thinking of moving this folder somewhere else, most likely under Kelly's work. I need to find/ make a lot of files for the data to be digitized and it might be better to have it more centralized. started looking the Address_ID script and my notes are below. I only got to line 22 before I got stuck, my notes are attached below. I will look into this more tomorrow and hopefully an epiphany will be made.  

[1989_Digitizaition_Notes.pdf](https://github.com/kbuzard/labs/files/9204792/1989_Digitizaition_Notes.pdf)

**Comment by kbuzard** [comment]
> I created a folder in the Python Script folder called 1989 where all of the copied scripts are, but I am thinking of moving this folder somewhere else, most likely under Kelly's work

I think the best options are to either integrate the 1989 work in with the 1998 as you've done already, OR to create a folder for everything from 1989 just inside ramosRivera. Then we would move Kelly's work on it into there. This would make more sense for someone looking at the file structure for the first time--they wouldn't have any reason to know what Kelly's folder is about.

Your documentation on this looks GREAT! I'm beginning to like using .md files directly on Github for things like documentation--then you don't have to have R open and keep knitting and attaching or pushing the file. If you want to try this, just go into the [heterogeneity folder](https://github.com/kbuzard/labs/tree/master/heterogeneity), click on "Add File" and then "Create New File." Give it a name at the top, and you can write in Markdown right there, and commit at the bottom of the page. There's a nice preview function that makes this work pretty well. The only drawback is that I don't think we've found a way to have a table of contents automatically generated like in Knitr. The other option we were exploring was making [Wiki pages](https://github.com/kbuzard/labs/wiki)--you could try that too if you want.

Some feedback I can give on the documentation, some of which might help you get going:
1. Put pngwork before Address_ID in the Order section
2. At the bottom of page 1, it's not clear that you've switched to talking about pngwork.py (I had to open up the two scripts to figure that out)
3. Labs1998.csv came from the work on the published paper. I can get something similar together for 1989, but we don't need it right now, so let's ignore it. This is safe to do because, although Antonio reads in in here, he never uses it
   - Line 18 gives the path the name "BuzzardPoints" (grr)
   - Line 22 then reads it into the object readBPoints
   - When I search both pngwork and Address_ID, readBPoints is never found again.
   - Bottom line: you can comment out these two lines, and put in an a comment saying that you don't know why it was read in here, as it was never used.
4. field.dta seems to be the same story as Labs1998.csv
5. You can create the equivalent of cattLabs97 by altering line 25 -- make it select 1989 instead of 1997
6. Next you need some background onCattell_corr_list.dta. Antonio found some mistakes in Ivan Png's original data. Png had an RA fix it and emailed us this file. You should be able to just pull out the 1989 data the same as for cattLabs97 (see immediately above). Please add this information into your notes so that it's documented. You can add that it was emailed to us on 2/14/2021 by Xi Xiong (xx646@nyu.edu)

**Comment by Kirs10-Riley** [comment]
Notes are attached below. I will created a .md file tomorrow. I was able to create cattLabs89 but the contents are empty. I figured out why and tried to figure out a way to create 89 versions of cattell-all and field and documented those trials  under pngwork.py outcome. 

[1989_Digitizaition_Notes.pdf](https://github.com/kbuzard/labs/files/9238220/1989_Digitizaition_Notes.pdf)

**Comment by kbuzard** [comment]
@Kirs10-Riley , the wiki looks great! I'm going to put my comments on it here

- In the "Lines 16-18" section, can you smooth this out a little? It looks like you just went back in an added the updates that we figured out, but a reader has to go through all the previous notes to finally get to that. It makes sense to keep the other stuff that you figured out in there, but this would be easier to read if that stuff were toward the end, made smaller, etc. You want the first thing someone reads to be the essential part, e.g. "We commented this line out because it's never used again in the script."
- the part about cattLabs89 under Address_ID needs to be updated--it's not longer empty, right?

**Comment by Kirs10-Riley** [comment]
@kbuzard I updated the notes

**Comment by kbuzard** [comment]
@Kirs10-Riley Here's an outline of the python code step I think you need to develop next in order to strip the facility names from the text that contains the names and addresses.

It's a for loop like the others you've used, but I recommend starting it with
` for x in len(Fac_Length)`
This will make *x* just a number that indexes each row in FacList (instead of being the text in the row itself).
- The advantage is that, since catFacilities lists all the facilities in the same order as FacList, you can use *x* to pull out the corresponding row from catFacilities

You'll want to start with something like
` y = catFacilities(x)`
Start out by just testing this, that is, close the loop with something like
`address += [y]`
and look at the output to make sure it makes sense.
- To do this, you'll need a step before for the for loop starts to create the output object: `address = [ ]` 

Now, you need to index into Fac_Length to get the length, so something like
`z = Fac_length(x)`
Again, test this by changing `address += [y]` to `address += [z]` and checking that you're getting a list of the lengths.

Now, write a line of code like [what you found in your internet search](https://stackoverflow.com/questions/11806559/removing-first-x-characters-from-string). I'm not sure exactly what it'll look like, but it will probably be something along the lines of
`q = catFacilities[z+2:]`
Once you've check that this works (again, reassigning q to `address`, see if you can directly assign `catFacilities[z+2:]` to `address`.

Once this is working, look at the output in `address` and think about how to strip out any bits like "Inc" that are left at the beginning of the string.
